src/uct/api/uct.h

```diff
@@ -519,6 +519,10 @@ enum uct_flush_flags {
                                                  error state, and it becomes
                                                  unusable for send operations
                                                  and should be destroyed. */
+    UCT_FLUSH_FLAG_REMOTE   = UCS_BIT(1)    /**< Guarantees that the target
```

Comment:
I think something along the lines: `Guarantees that all previous UCP memory update operations (put and  atomics) are completed - the target memory was updated  and updated memory is visible for all processing elements in the system`. @tonycurtis Any thoughts ?

Reply:
@tonycurtis any comments?

Reply:
Slightly updating it - `Guarantees that all previous UCP memory update operations (put, atomics, etc.) are completed, the target memory of these operation was updated, and the updated memory is visible for all processing elements in the system.`

Reply:
@brminich It is actually very generic term. Arm specification, x86, and many other use the term to describe compute elements in the system. We cannot use "endpoints" because the intent here to make the memory visible outside of scope of UCX library. Anyways, I'm open for any other suggestions.

Reply:
@tonycurtis , can you please review ?

Reply:
Sorry, gtihub invalidated my email address until I did something, for some reason.

You could use "ranks" instead of PEs and I think everyone would understand what is meant (Cray/HPE still do this conflation).  Pasha's orig comment looks ok to me

Reply:
We don't have ranks definition anywhere.  UCX is used a lot outside of MPI/SHMEM so I would avoid using ranks.

Reply:
How about
```
Guarantees that all previous remote memory update operations on the endpoint @a ep (put and atomics) 
are completed and the updated memory contents are visible to the remote process the endpoint is connected to.
```

Reply:
@yosefe  the way I read `the remote process the endpoint is connected` that it is visible for other EP on the network, which is relatively weak guarantee, since does not require flush to the memory. 
I was about to suggest the following:
`Guarantees that all previous remote memory update operations on the endpoint @a ep (put and atomics) 
are completed and the updated memory contents are visible to all processes (remote and local) on the target @a ep.`
But `process` is a software construct and for example the update may not be visible if in a memory if it is not shared with `all` processes. `Processing elements` is a hardware construct which actually means that any CPU core or and accelerator that has connection to memory bus (like GPU) can actually access the memory. Sorry for been stickler on this.

Reply:
I'm a bit confused by "all processes (remote and local) on the target ep". Maybe "... and the updated memory contents are globally visible" which implies any hardware that can access the memory?

Reply:
`global visible for all processing elements in the system` ?

Reply:
sounds good to me

Reply:
@hoopoepg I think we have a consensus.

Reply:
@hoopoepg pls update according to the latest comments and let's also get Tony's approval

Reply:
done